<!-- Developer: Sanha Lee -->
<!-- Date: April.13.2020 -->
<!-- Description: This is a launch file that runs all nodes for generating 2D walking gait in Gazebo simulation -->

<launch>


    <node pkg ="openDog_description" type="new_foot_path.py" name="pub_foot_path_1">
        <param name = "foot_position" value = "/footPosition_1"/>
        <param name = "phase_shift" value = '0'/>
        <param name = "y_center" value = '25'/>
        <param name = "y_lift" value = '5.5'/>
        <param name = "leg_pace" value = '3.0'/>
        <param name = "x_stride" value = '3.3'/>
        <param name = "x_center" value = '10.0'/>
    </node>

    <node pkg ="openDog_description" type="new_foot_path.py" name="pub_foot_path_2">
        <param name = "foot_position" value = "/footPosition_2"/>
        <param name = "phase_shift" value = '3.1415926'/>   <!-- '0.785398' -->
        <param name = "y_center" value = '25'/>
        <param name = "y_lift" value = '5.5'/>
        <param name = "leg_pace" value = '3.0'/>
        <param name = "x_stride" value = '3.3'/>
        <param name = "x_center" value = '10.0'/>
    </node>

    <node pkg ="openDog_description" type="new_foot_path.py" name="pub_foot_path_3">
        <param name = "foot_position" value = "/footPosition_3"/>
        <param name = "phase_shift" value = '4.712389'/>    <!-- '2.356194'/> -->
        <param name = "y_center" value = '25'/>
        <param name = "y_lift" value = '3'/>
        <param name = "leg_pace" value = '3.0'/>
        <param name = "x_stride" value = '3.3'/>
        <param name = "x_center" value = '10.0'/>
    </node>

    <node pkg ="openDog_description" type="new_foot_path.py" name="pub_foot_path_4">
        <param name = "foot_position" value = "/footPosition_4"/>
        <param name = "phase_shift" value = '1.570796'/>      <!-- '1.570796'/> -->
        <param name = "y_center" value = '25'/>
        <param name = "y_lift" value = '3'/>
        <param name = "leg_pace" value = '3.0'/>
        <param name = "x_stride" value = '3.3'/>
        <param name = "x_center" value = '10.0'/>
    </node>

    <node pkg="openDog_description" type="2D_inverse_kinematics.py" name="inverse_kinematics_1">
        <param name = "femur_angle" value = "/theta_f_1"/>
        <param name = "tibia_angle" value = "/theta_t_1"/>
        <param name = "hip_angle" value = "/theta_h_1"/>
        <param name = "foot_position" value = "/footPosition_1"/>
    </node>




    <node pkg="openDog_description" type="2D_inverse_kinematics.py" name="inverse_kinematics_2">
        <param name = "femur_angle" value = "/theta_f_2"/>
        <param name = "tibia_angle" value = "/theta_t_2"/>
        <param name = "hip_angle" value = "/theta_h_2"/>
        <param name = "foot_position" value = "/footPosition_2"/>
    </node>

    <node pkg="openDog_description" type="2D_inverse_kinematics.py" name="inverse_kinematics_3">
        <param name = "femur_angle" value = "/theta_f_3"/>
        <param name = "tibia_angle" value = "/theta_t_3"/>
        <param name = "hip_angle" value = "/theta_h_3"/>
        <param name = "foot_position" value = "/footPosition_3"/>
    </node>

    <node pkg="openDog_description" type="2D_inverse_kinematics.py" name="inverse_kinematics_4">
        <param name = "femur_angle" value = "/theta_f_4"/>
        <param name = "tibia_angle" value = "/theta_t_4"/>
        <param name = "hip_angle" value = "/theta_h_4"/>
        <param name = "foot_position" value = "/footPosition_4"/>
    </node>



    <node pkg="openDog_description" type="gazebo_joint.py" name="gazebo_joint_1">
        <param name = "femur_angle" value = "/theta_f_1"/>
        <param name = "tibia_angle" value = "/theta_t_1"/>
        <param name = "tibia_controller" value = "/opendog/joint1_position_controller/command"/>
        <param name = "femur_controller" value = "/opendog/joint5_position_controller/command"/>
    </node>

    <node pkg="openDog_description" type="gazebo_joint.py" name="gazebo_joint_2">
        <param name = "femur_angle" value = "/theta_f_2"/>
        <param name = "tibia_angle" value = "/theta_t_2"/>
        <param name = "tibia_controller" value = "/opendog/joint2_position_controller/command"/>
        <param name = "femur_controller" value = "/opendog//joint6_position_controller/command"/>
    </node>

    <node pkg="openDog_description" type="gazebo_joint.py" name="gazebo_joint_3">
        <param name = "femur_angle" value = "/theta_f_3"/>
        <param name = "tibia_angle" value = "/theta_t_3"/>
        <param name = "tibia_controller" value = "/opendog//joint3_position_controller/command"/>
        <param name = "femur_controller" value = "/opendog/joint7_position_controller/command"/>
    </node>

    <node pkg="openDog_description" type="gazebo_joint.py" name="gazebo_joint_4">
        <param name = "femur_angle" value = "/theta_f_4"/>
        <param name = "tibia_angle" value = "/theta_t_4"/>
        <param name = "tibia_controller" value = "/opendog/joint4_position_controller/command"/>
        <param name = "femur_controller" value = "/opendog/joint8_position_controller/command"/>
    </node>


</launch>